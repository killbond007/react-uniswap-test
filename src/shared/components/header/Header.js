import { Button, Dropdown, Menu } from 'antd'
import { DownOutlined, UserOutlined } from '@ant-design/icons'
import { useDispatch, useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import React, { useCallback } from 'react'

import { changeAccount, connectWallet, disconnectWallet } from 'modules/root/actions/rootActions'

import * as Styled from './__styles__/Header.styles'

const Header = () => {
	const { web3 } = useSelector((state) => state?.root)
	const { t } = useTranslation('root')

	const dispatch = useDispatch()

	const _onConnectClick = useCallback(() => dispatch(connectWallet()), [dispatch])

	const _onChangeAccountClick = useCallback(() => dispatch(changeAccount()), [dispatch])

	const _onDisconnectClick = useCallback(() => dispatch(disconnectWallet()), [dispatch])

	const address = web3?.currentProvider?.accounts?.[0] || web3?.currentProvider?.selectedAddress

	return (
		<Styled.Root>
			<Menu mode="horizontal" defaultSelectedKeys={['2']}>
				<Menu.Item key="1">nav 1</Menu.Item>
				<Menu.Item key="2">nav 2</Menu.Item>
				<Menu.Item key="3">nav 3</Menu.Item>
			</Menu>
			{address ? (
				<Dropdown
					overlay={
						<Menu>
							<Menu.Item onClick={_onDisconnectClick}>Disconnect</Menu.Item>
							<Menu.Item onClick={_onChangeAccountClick}>Change Account</Menu.Item>
						</Menu>
					}
				>
					<a>
						<UserOutlined /> {address.slice(0, 5)}...${address.slice(-5)} <DownOutlined />
					</a>
				</Dropdown>
			) : (
				<Button onClick={_onConnectClick} type="primary">
					{t('root:connectButton')}
				</Button>
			)}
		</Styled.Root>
	)
}

export default Header
