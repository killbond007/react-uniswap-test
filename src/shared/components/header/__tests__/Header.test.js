import { LocationProvider, createHistory, createMemorySource } from '@reach/router'
import { Provider } from 'react-redux'
import { createStore } from 'redux-dynamic-modules'
import { render } from '@testing-library/react'
import React from 'react'

import Header from 'shared/components/header/Header'

const initialState = {}
const getStore = (state) => createStore({ initialState: state })
const getInstance = (state = initialState) => (
	<LocationProvider history={createHistory(createMemorySource('/'))}>
		<Provider store={getStore(state)}>
			<Header />
		</Provider>
	</LocationProvider>
)

describe('Header', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})
})
