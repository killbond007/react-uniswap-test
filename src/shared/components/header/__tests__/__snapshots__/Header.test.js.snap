// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: primary.9 !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 12px !important;
}

.c1:hover {
  cursor: pointer;
}

.c2 .ant-badge-count {
  background-color: primary.6;
}

<header
    class="ant-layout-header c0"
  >
    <img
      alt="root:header.aria.logo"
      class="c1"
      src="logo-origami.svg"
    />
    <ul
      class="ant-menu ant-menu-dark ant-menu-root ant-menu-horizontal"
      role="menu"
      style="background-color: rgb(2, 36, 74);"
    >
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
        role="menuitem"
      >
        root:header.menu.studies
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        root:header.menu.toolbox
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        root:header.menu.settings
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        <span
          class="ant-badge c2"
        >
          <img
            alt="root:header.aria.notification"
            src="notification.svg"
          />
        </span>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        <div
          class="ant-dropdown-trigger"
        >
           
        </div>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="visibility: hidden; position: absolute; display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
  </header>
</DocumentFragment>
`;
